### ISSUE TITLE
[L] PRDT: Prepare Search for Use

### ISSUE THREAD

**Issue by Dianadec** [bcgov/reserve-rec-public#14]
#### Description:
Informed by the "choose your own adventure" concept from story mapping, we need to explore how we might optimize search, specific to backcountry booking. Update from Aug 20 refinement - eg: you don't need to state your dates or what you're looking for precisely. more like, you can look at a map and see everything and not necessarily have to specify the type you want.

This ticket is to explore and identify next steps for **public facing search**. When this ticket is complete we should be ready to hook-up open search to the first iteration of the front-end UI (FE to be built out after designs are ready).

#### Acceptance Criteria:

- [ ] Review output from bcgov/reserve-rec-api#6 data exploration and bcgov/reserve-rec-api#10 policy inventory
- [ ] Document considerations/limitations specific to backcountry search that should be incorporated into search designs
   - [ ] Performance considerations
   - [ ] Scalability considerations
- [ ] Determine best approach for data and set filters for opensearch (start with just basic filters/search queries)
  - [ ] basic queries/parameters are created in open search and it is ready to be hooked up to first iteration of the FE UI (FE is not yet built) 

#### Development Checklist:

- [x] All data authoritatively is in dynamo with exception of geojson
- [ ] DynamoStream should replicate changes to auth data to OpenSearch
- [ ] searching opensearch is for map or any unstructured query
- [ ] Dates open/close should be pulled via script/offline into the system, and an ETL to update DynamoDB/OS data so that searching endpoints will work. No live connections to strapi should be made.

#### Dependencies

- Blocked by bcgov/reserve-rec-api#6 
- Blocking bcgov/reserve-rec-public#13 

**Relevant documentation as reference**

**Definition of Ready**

- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [ ] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:

![Image](https://github.com/bcgov/parks-data-register/assets/35785153/b038214f-5d59-48c5-becc-01c742adee98)

-

**Comment by Dianadec** [comment]
Refinement notes:
- we have everything we need technically to do the search piece
- we have the raw data to create an experience but someone will have to go through it and grab all the segments that apply to create a whole thing around the experience. Eg. West coast trail. That's a huge data exercise. And, need to build a system that looks at all the inventory and notes which items are now limited on a resource by resource instance. Todo: figure out how things are attributed together. Secondary is building a tool that lets them manage that.
- backcountry registrations don't have a limit of permits they will give you (limit in system though - very high) - but you could go the day-of and get it. Reservations IS limited. But some registrations get changed into a reservation to limit the capacity, so that people can also be guaranteed a tent pad. Registrations = tent pads are fcfs. Every year reservations systems will do a call-out. First, an RSO, area supervisor section head, then Regional Director. Executive - approves decision note. Overall res services wants to review this process.

**Comment by Dianadec** [comment]
Notes from Darcie/Mark/Diana mtg:

less about designs - more about all the questions
- Do we want this recorded to make sense of the story?
- Defining the user experience (not about the Front end look), but how are they moving through the system conceptually. 

-->To find the questions, let's look at the user story map. Eg: give me user story #1 and let me try to conceptualize at each stage of that person's flow, what info we'll be needing to grab, eg: looking at park or looking at geo. If geo, what are they filtering on (food caches, region, what, etc). Then want to see where everything is on a map, want to go from point A to Point B - what are my requirements? Do I need to register for this area, reserve? (this brings in policy, biz rules, ). TLDR need to go through a story, walk through it from a data perspective, then find the data and find a group of data that wraps around it.
--> need Brittany or Jessica to link policies to geo

Does Cam's flow diagram capture any of this? yes, and we need to build upon it.

**Comment by Dianadec** [comment]
Aug 20 Refinement notes:
- This is pretty much done (the exploration piece) - opensearch is built and running
-

**Comment by JLWade** [comment]
Sept 19th Refinement Notes:
- ticket is ready to go, just needs estimates
- for data looking at all parks so we can test performance and scalability with opensearch (rather than focus on the 3 parks with backcountry)
- filters will be basic and then updated based on user research
- result of this ticket will be that opensearch is ready to be hooked up to a FE map and text search (FE is covered in another ticket)

**Comment by Dianadec** [comment]
Sprint Planning notes:
- The first queries we are going to set will enable a user to search for a park name and/or cam's zones.

**Comment by Dianadec** [comment]
Need a list of the ways we need to filter things (in plain English) - eg: Strathcona protected area vs. all things with Strathcona in the name.  @LindsayMacfarlane -

**Comment by manuji** [comment]
Tested on Dev: **Passed**
* verified that I can search for any keyword and if there is matching park names, the results displayed on the FE. 
* Passing to PO review.

**Comment by LindsayMacfarlane** [comment]
Tested on Test. 

- Can search for park names and have results displayed in left hand column (parks I searched for included: Strathcona, Garibaldi, Goldstream, Rathtrevor, boat, picnic).

Observation, which I believe is outside the scope of this ticket:
- No pin placed on map for Strathcona
- Pins didn't clear between searches (unclear if that is the expected behaviour at this time)

Thank you for your work on this one @danieltruong - it's looking good! And thank you for your testing @manuji 

@pavelcreates - flagging feedback on pin behaviour for design input in future.
